commit 191802d45b366cca414a65e0c4ebe8c4fd41bd82
Author: aragaer <aragaer@gmail.com>
Date:   Sat Mar 14 22:41:03 2020 +0300

    fix: Reconnect failure
    
    The "notified on disconnect" flag prevents double disconnects but once
    it is set it is not being reset. So it does prevent double disconnect
    event on first disconnect but on second disconnect it doesn't fire
    "disconnected" event at all. This fix resets the "notified on
    disconnect" after skipping the "disconnected" event.
    
    closes issue #294

diff --git a/stomp/transport.py b/stomp/transport.py
index 2240af8..c31c58e 100644
--- a/stomp/transport.py
+++ b/stomp/transport.py
@@ -360,6 +360,7 @@ class BaseTransport(stomp.listener.Publisher):
                 self.notify("disconnected")
             with self.__connect_wait_condition:
                 self.__connect_wait_condition.notifyAll()
+            self.__notified_on_disconnect = False
 
     def __read(self):
         """
